str/associatr/get_cis_numpy_files.py

```diff
@@ -0,0 +1,248 @@
+        )
+        gene_pheno = gene_pheno[['sample_id', 'gene_inverse_normal']]
+
+        gene_pheno_cov = gene_pheno.merge(covariates, on='sample_id', how='inner')
+
+        # filter for samples that were assigned a CPG ID; unassigned samples after demultiplexing will not have a CPG ID
+        gene_pheno_cov = gene_pheno_cov[
+            gene_pheno_cov['sample_id'].str.startswith('CPG')
+        ]
+
+        # add in cohort as a covariate
+        cohort_cpg_id = adata.obs[['cpg_id', 'cohort']]
+        mapping = {'TOB': 1, 'BioHEART': 2}
+        cohort_cpg_id['cohort'] = cohort_cpg_id['cohort'].map(mapping)
+        cohort_cpg_id.rename(columns={'cpg_id': 'sample_id'}, inplace=True)
+        cohort_cpg_id = cohort_cpg_id.drop_duplicates()
+
+        gene_pheno_cov = gene_pheno_cov.merge(
+            cohort_cpg_id, on='sample_id', how='inner'
+        )
+
+        gene_pheno_cov['sample_id'] = gene_pheno_cov['sample_id'].str[
+            3:
+        ]  # remove CPG prefix because associatr expects id to be numeric
+
+        gene_pheno_cov['sample_id'] = gene_pheno_cov['sample_id'].astype(float)
+
+        gene_pheno_cov = gene_pheno_cov.to_numpy()
+        with hl.hadoop_open(
+            output_path(
+                f'pheno_cov_numpy/{version}/{cell_type}/{chromosome}/{gene}_pheno_cov.npy'
+            ),
+            'wb',
+        ) as f:
+            np.save(f, gene_pheno_cov)
+
+
+@click.option('--input-h5ad-dir', help='GCS directory to the input AnnData objects')
+@click.option(
+    '--input-pseudobulk-dir', help='GCS directory to the input pseudobulk CSV files'
+)
+@click.option('--input-cov-dir', help='GCS directory to the input covariate CSV files')
+@click.option('--chromosomes', help=' eg chr22, comma separated if multiple')
+@click.option('--cell-types', help='cell type, comma separated if multiple')
+@click.option('--cis-window', help='cis window size in bp')
+@click.option('--version', default='v1', help='version of the output files')
+@click.option('--job-cpu', default=2, help='Number of CPUs to use for the job')
+@click.option('--job-memory', default='standard', help='Memory to use for the job')
+@click.option('--job-storage', default='5G', help='Storage to use for the job')
+@click.option(
+    '--max-parallel-jobs',
+    type=int,
+    default=500,
+    help=('To avoid exceeding Google Cloud quotas, set this concurrency as a limit.'),
+)
+@click.option(
+    '--min-pct',
+    default=1,
+    help='filter out genes that are expressed in fewer than XX% of cells',
+)
+@click.option(
+    '--remove-samples-file',
+    default=None,
+    help='GCS path to the file containing the list of samples to remove',
+)
+@click.command()
+def main(
+    input_h5ad_dir,
+    input_pseudobulk_dir,
+    input_cov_dir,
+    chromosomes,
+    cell_types,
+    cis_window,
+    version,
+    job_cpu,
+    job_memory,
+    job_storage,
+    max_parallel_jobs,
+    min_pct,
+    remove_samples_file,
+):
```

Comment:
but if you do have arguments, could you please add type hinting
```suggestion
def main(
    input_h5ad_dir: str,
    input_pseudobulk_dir: str,
    input_cov_dir: str,
    chromosomes: str,  # but taken from config should be a list[str]
    cell_types: str,  # but taken from config should be a list[str]
    cis_window:  int,
    version: str,
    job_cpu: int,
    job_memory: str,
    job_storage: str,
    max_parallel_jobs: int,
    min_pct: int,
    remove_samples_file:  str | None,
):
```